benchopt/runner.py

```diff
@@ -396,14 +358,64 @@ def _run_benchmark(benchmark, solvers=None, forced_solvers=None,
         timeout=timeout, pdb=pdb, collect=collect
     )
 
-    results = parallel_run(
-        benchmark, run_one_solver, common_kwargs, all_runs,
-        config=parallel_config, collect=collect
+    run_one_to_cvg_cached = benchmark.cache(
+        run_one_to_cvg, ignore=['force', 'pdb'], collect=collect
     )
 
+    if collect:
+        _run_one_to_cvg_cached = run_one_to_cvg_cached
+
+        def run_one_to_cvg_cached(**kwargs):
+            res = _run_one_to_cvg_cached(**kwargs)
+            key = (
+                kwargs['meta']['dataset_name'],
+                kwargs['meta']['objective_name'],
+                kwargs['meta']['solver_name']
+            )
+            return res if res is not None else ([], (key, 'not run yet'))
+
+    def run_one_to_cvg_final(**kwargs):
+        results = None
+        try:
+            results = run_one_to_cvg_cached(**kwargs)
+        except FailedRun as e:
+            key = (
+                kwargs['meta']['dataset_name'],
+                kwargs['meta']['objective_name'],
+                kwargs['meta']['solver_name']
+            )
+            results = ([], (key, e.status))
+        return results
+
+    total_cvg_kwargs = []
+    for run_kwargs in all_runs:
+        cvg_kwargs = get_run_args(
+            **common_kwargs, **run_kwargs
+        )
+        total_cvg_kwargs.extend(cvg_kwargs)
+
     run_statistics = []
-    for curve in results:
-        run_statistics.extend(curve)
+
+    results_generator = parallel_run(
+        benchmark, run_one_to_cvg_final, total_cvg_kwargs,
+        config=parallel_config, collect=collect
+    )
+
+    status_dict = {}
+
+    for results in results_generator:
+        result = results[0]
+        run_statistics.extend(result)
+
+        key, status = results[1]
+        if key not in status_dict:
+            status_dict[key] = status
+        elif status_dict[key] == 'done':
+            status_dict[key] = status
+
+    for key, status in status_dict.items():
+        terminal.set(dataset=key[0], objective=key[1], solver=key[2])
+        terminal.show_status(status)
```

Comment:
The issue here is that the status should be updated as soon as we have all the info from one solver run.